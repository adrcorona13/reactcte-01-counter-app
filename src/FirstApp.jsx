import PropTypes from 'prop-types';

// export const FirstApp = (props) => {
export const FirstApp = ({ 
  title, 
  subtitle, 
  value,
}) => {
  
  // if (!title) {
  //   throw new Error('El titulo no existe')
  // }

  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{value}</p>
    </>
  );
};


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  value: PropTypes.number
}

FirstApp.defaultProps = {
  title: "No hay titulo",
  subtitle: "No hay subtitulo",
  value: 0
}