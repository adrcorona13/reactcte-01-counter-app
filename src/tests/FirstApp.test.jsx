import { render } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en FirstApp', () => { 
    test('debe de hacer match con el snapshot', () => { 
        const title = 'titulo';
        const { container } = render(<FirstApp title={title}/>);
        expect( container ).toMatchSnapshot();
        // console.log(container);
     })

     test('debe de mostrar el titulo en un h1', () => { 
        const title = 'titulo';
        const { container, getByText } = render(<FirstApp title={title}/>);
        expect( container ).toMatchSnapshot();
        expect( getByText(title) ).toBeTruthy();
        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);
     })
 })   