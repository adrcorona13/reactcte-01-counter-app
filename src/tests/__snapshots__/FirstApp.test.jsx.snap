// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Adrian
  </p>
</div>
`;

exports[`Pruebas en FirstApp debe de mostrar el titulo en un h1 1`] = `
<div>
  <h1>
    titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Adrian
  </p>
</div>
`;
